### DISCUSSION TITLE
Battery discharge schedule and reserve

### DISCUSSION THREAD

**Discussion by Mougli** [WillCodeForCats/solaredge-modbus-multi#622] [Q&A]
Hello,

I have an electricity tariff where import is cheap at night and export is high in the late afternoon.

I want to export as much energy as possible during that peak time, but also want to make sure I keep enough in the battery to see me through the night. And if the battery does run out, I want it to charge from the grid only during the cheap hours and use that stored energy for the rest of the night.

The solarEdge app can do that scheduling, but doesn't seem to have any setting to ensure a minimum state of charge for the battery.

I wanted to double check if the following HA setup is correct for my goal:

- Set Storage control Mode to Remote control
- Set AC Charge Limit to X (where X = the point at which I think I've got enough in the battery to last the night)
- Set Backup Reserve to Y% (where Y = the point below which the inverter should not export to the grid)
- Set Discharge Limit to what's been agreed with the utility

Then setup automations:

a. During high export window, change storage command mode to maximise export (backup reserve will ensure I don't drop past my limit)
b. During low import window, if battery below a certain level, allow AC charging (AC charge limit will handle the cut-off) and change storage command to charge from solar and grid.
c. At any other time, stick to the default (no AC charging, and storage command set to maximize self-consumption).

Actually as I write this it seems like my automations would only need to update the storage command depending on time of day, and I wouldn't need to adjust AC charging settings? Is my understanding correct?

Thanks!

**Comment by Ashpork** [comment]
Hi,

For during the summer, I'm on a similar tariff to your description.

Potentially, yes that could work. Personally I haven't tried any of the AC Charge Limit functionality, and have typically found the "Backup Reserve" to not be respected by my system. I previously just managed SoC levels using automations and scripts.

As an alternative option, I now use [Predbat](https://github.com/springfall2008/batpred) to manage my charge and discharge schedules, and it does this based on solar forecasting and average loads. 

There are certainly some limitations with Predbat and the SolarEdge combination, in that Predbat was initially designed for GivEnergy systems and is well geared towards those set ups, but after some tinkering it works fairly well for me. They do have some template configurations to help it work with SolarEdge systems, but I found I had to go a little further to make it completely sensible. It's limitation is in that it is largely limited to either putting my system into a discharge, charge or Maximum Self Consumption mode. Predbat with a GivEnergy solution will do some finer control on the inverter, but for the most it does enough for SolarEdge systems.

Might be worth playing around with that. On the face of it, it's quite an involved setup, but gets easier the more you read through their help documents and guides.

**Reply by Mougli** [reply]
Hey, thanks for the reply, I might take a look at Predbat, it does sound potentially useful.

And yeah I figured I could handle SoC with automations, but thought I'd ask about Backup Reserve as the documentation has no description for it.

**Comment by deqw** [comment]
> The solarEdge app can do that scheduling, but doesn't seem to have any setting to ensure a minimum state of charge for the battery.

The setting you search for is in the SolarEdge app when you have a Backup Interface installed.

Edit: resized the image

![IMG_8213](https://github.com/user-attachments/assets/9a871214-169a-436e-9429-5c24f5a3fc13)

**Reply by Mougli** [reply]
Oh, I didn't know you needed an additional device to enable backup. That explains when my power went out during a neighbourhood outage even though I had full battery. Wish my installer mentioned this...

**Reply by deqw** [reply]
Yep the Backup Interface is needed for that. There is still a one second hick-up in power when grid goes down when you have a 3-phase inverter (for the 1-phase inverter there was already a firmware update that sorts this out). When the grid power returns the transition is seamless. For all the network stuff like router and switches and of course the RPi that runs HA I use an additional small UPS to overcome the hick-up.

**Comment by deqw** [comment]
I do my automations in HA//Node-Red which gives me better oversight than the HA-Automations. I do not discharge my battery to the grid for purpose to gain some money but do grid charging on schedule when tariff is ok or when needed. Basically Storage Control Mode is in 'Remote Control' all the time, then create a bunch of Helpers in HA and program some code in Node-Red. A panel in HA to control does the rest. With Node-Red you can also do all the inverter power control (e.g. limit Power to X% depending on grid or inverter V-AC to prevent stopping the inverter for high voltage). 

![Screenshot 2024-07-13 at 11 14 25](https://github.com/user-attachments/assets/ef87f0a5-6a9f-43c5-b42f-a85826e892cf)

**Reply by Mougli** [reply]
I have not touched Node-Red at all, maybe it's time to take a look.

**Reply by deqw** [reply]
In the end this can take a lot of time to figure out how to set this up. To get you started I've made an example for you which sketches just my approach. Find below the example json that you can simply import or copy/paste into Node-Red. I included an example how to set the Active Power Limit. In the example the power limit is set by an Input Node, but in real world I pass in the value by looking at incoming V-AC values and calculate what the power limit value should be. Consider putting all your triggering states on a separate flow and then link to the flow where you process the changes.

![Screenshot 2024-07-13 at 13 08 53](https://github.com/user-attachments/assets/7684028f-ed1d-444b-a540-5ec9942c53ff)

[example-solaredge-inverter-flow.json](https://github.com/user-attachments/files/16201906/example-solaredge-inverter-flow.json)

**Reply by Mougli** [reply]
Thanks for taking the time to do this, I'll look into it!
